### ISSUE TITLE
AQL query wrongfully returning no results on 3.6.1

### ISSUE THREAD

**Issue by Darkheir** [arangodb/arangodb#11051]
## My Environment

* __ArangoDB Version__:        3.6.1
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server

## Component, Query & Data

__Affected feature__:  AQL query

__AQL query (if applicable)__:

```
FOR item IN test  
    FILTER ( item.enabled == true AND (item.type == "a" OR item.type == "b"))
    SORT item.modified ASC
    LIMIT 0, 10
    RETURN item
```

__AQL explain (if applicable)__:

```
Query String (155 chars, cacheable: true):
 FOR item IN test  
     FILTER ( item.enabled == true AND (item.type == "a" OR item.type == "b"))
     SORT item.modified ASC
     LIMIT 0, 10
     RETURN item

Execution plan:
 Id   NodeType          Est.   Comment
  1   SingletonNode        1   * ROOT
 10   IndexNode            1     - FOR item IN test LET #3 = item.`modified`   /* persistent index scan */    FILTER (item.`enabled` == true)   /* early pruning */   /* with late materialization */
  6   SortNode             1       - SORT #3 ASC   /* sorting strategy: standard */
  7   LimitNode            1       - LIMIT 0, 10
 12   MaterializeNode      1       - MATERIALIZE item
  8   ReturnNode           1       - RETURN item

Indexes used:
 By   Name                      Type         Collection   Unique   Sparse   Selectivity   Fields                   Ranges
 10   idx_1657696213696249856   persistent   test         false    false       100.00 %   [ `type`, `modified` ]   (item.`type` in [ "a", "b" ])

Optimization rules applied:
 Id   RuleName
  1   move-calculations-up
  2   move-filters-up
  3   move-calculations-up-2
  4   move-filters-up-2
  5   replace-or-with-in
  6   use-indexes
  7   remove-filter-covered-by-index
  8   move-filters-into-enumerate
  9   late-document-materialization
```

__Dataset__:

```json
[
  {
    "_key": "138280768",
    "_id": "test/138280768",
    "_rev": "_a_SdjuW--_",
    "type": "a",
    "enabled": true,
    "modified": "2019-10-11T14:04:15.77685Z"
  },
  {
    "_key": "138280793",
    "_id": "test/138280793",
    "_rev": "_a_Sdz3a--_",
    "type": "b",
    "enabled": true,
    "modified": "2019-10-11T14:04:15.77685Z"
  }
]
```

## Steps to reproduce

1. Test collection with the 2 items above
2. Add a perisstent index on `type, modified`
3. Run the query above

__Problem__: The query returns no results and should return the 2 documents

__Expected result__: Both of the documents are returned

__Additional Informations__:

* The query without the index works
* The query without the `LIMIT` works
* The query without the `SORT` works
* The query without `item.enabled == true` in the `FILTER` part works
* Starting ArangoDB with `--query.optimizer-rules "-late-document-materialization"` also fixes the issue

**Comment by jsteemann** [comment]
Fixed via the above PRs. Target version for the bugfix is 3.6.2.
Thanks for reporting the issue!

**Comment by Darkheir** [comment]
That is awesome ! Do you know when 3.6.2 version is planned for release ?

**Comment by Simran-B** [comment]
There is not ETA yet, but it should only be a few weeks away.

BTW. is https://github.com/arangodb/arangodb/pull/11092 also related?

**Comment by maxkernbach** [comment]
Hi @Darkheir,

Version 3.6.2, which includes a fix for this issue, has just been released.

From the [Changelog](https://github.com/arangodb/arangodb/blob/15ce60f850ed2f8ea9bb861eb8160e7bc9e203e5/CHANGELOG#L85):
> * Fixed issue #11051: AQL query wrongfully returning no results on 3.6.1.
> 
>   This change prevents the "late-document-materialization" optimizer rule from
>   kicking in in cases when an index scan also uses early pruning of index
>   entries. In this case, the late document materialization will lead to the
>   filter condition used in early pruning not being applied correctly, potentially
>   producing wrong results.

Version 3.6.2 distributives can be obtained at the links below:
- https://www.arangodb.com/download/
- https://download.arangodb.com/arangodb36/

**Comment by Darkheir** [comment]
Awesome, thank you for your work!
